web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/util/StorageHelper.java

```diff
@@ -101,18 +104,20 @@ public JobMetadata getJobMetadata(String jobId) {
    *
    * @param jobId
    * @param url
+   * @param validatorVersion
    * @throws Exception
    */
-  public void saveJobFileFromUrl(String jobId, String url) throws Exception {
-    // Read file into memory
-    var urlInputStream = new BufferedInputStream(new URL(url).openStream());
-
-    // Upload to GCS
-    var blobId = BlobId.of(USER_UPLOAD_BUCKET_NAME, jobId + "/" + jobId + ".zip");
-    var mimeType = "application/zip";
-    var blobInfo = BlobInfo.newBuilder(blobId).setContentType(mimeType).build();
-    var fileBytes = urlInputStream.readAllBytes();
-    storage.create(blobInfo, fileBytes);
+  public void saveJobFileFromUrl(String jobId, String url, String validatorVersion)
+      throws Exception {
+    var blobId = BlobId.of(USER_UPLOAD_BUCKET_NAME, jobId + "/" + FILE_NAME);
+    var blobInfo = BlobInfo.newBuilder(blobId).setContentType("application/zip").build();
+    URL signedURL =
+        storage.signUrl(
+            blobInfo, 1, TimeUnit.HOURS, Storage.SignUrlOption.httpMethod(HttpMethod.POST));
+    try (WriteChannel writer = storage.writer(signedURL)) {
+      OutputStream outputStream = Channels.newOutputStream(writer);
```

Comment:
Using the Channels output stream reduces the memory footprint by avoiding loading the full file in memory.